### PR TITLE
feat(egctl): add support for egctl to translate from gateway-api resources to IR

### LINKED ISSUE #2798: Enhance "egctl translate" to support translating Gateway-API resources to IR
*Description*:
Having the ability to see what the Envoy Gateway IR for a given set of Gateway API resources shortens development cycles when trying to figure out why something isn't translated correctly.

### PR DESCRIPTION
**Pull request by liorokman**
**What this PR does / why we need it**:

Adds support for translating Gateway-API resources to IR.

**Which issue(s) this PR fixes**:
Fixes #2798

### CODE REVIEW

File: internal/cmd/egctl/translate.go [hunk 1]
```diff
@@ -93,6 +96,9 @@ func newTranslateCommand() *cobra.Command {
   # Translate Gateway API Resources into All xDS Resources in YAML output,
   # also print the Gateway API Resources with updated status in the same output.
   egctl experimental translate --from gateway-api --to gateway-api,xds --type all --output yaml --file <input file>
+
+  # Translate Gateway API Resources into IR in YAML output,
```
Comment (shawnh2) [inline internal/cmd/egctl/translate.go:100]: can you also update the user doc? https://gateway.envoyproxy.io/latest/user/egctl/#egctl-experimental-translate thx

**Review by arkodg — DISMISSED** [review]
nice !

**Review by shawnh2 — APPROVED** [review]
lgtm thx

**Review by cnvergence — APPROVED** [review]
awesome

**Review by arkodg — APPROVED** [review]

**Pull request merged**